src/main/java/com/ait/lienzo/client/core/shape/wires/layout/direction/DirectionLayout.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape.wires.layout.direction;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.ait.lienzo.client.core.shape.wires.layout.label.LabelLayout;
+import com.ait.lienzo.client.core.shape.wires.layout.label.LabelLayout.Builder;
+
+public class DirectionLayout
+{
+    public interface Direction
```

Comment:
is there some real need for using this `Direction` interface with no exposed methods, is it kinda marker interface?  mmm probably it can be removed... I mean, if it does not expose any method, it just can be replaced by a generic type, so what about refactoring as  `public class DirectionLayout<L> { ... }`, also inheritance could be removed for `HorizontalAlignment` and `VerticalAlignment` as well, does that makes sense? WDYT?

Reply:
mmm, I'm using this interface just to identify the enums, `VerticalAlignment` and `HorizontalAlignment` when they are used to set margins in a generic way as a key of a map,  `Map<Direction, Double>`. So in fact currently the interface does not have any specific method, any suggestion?